drivers/dma/dma_stm32.c

```diff
@@ -97,14 +97,15 @@ static void dma_stm32_irq_handler(const struct device *dev, uint32_t id)
 	}
 
 	/* the dma stream id is in range from STREAM_OFFSET..<dma-requests> */
-	if (dma_stm32_is_tc_active(dma, id)) {
-		dma_stm32_clear_tc(dma, id);
+	if (dma_stm32_is_ht_active(dma, id)) {
+		stream->dma_callback(dev, stream->user_data, callback_arg, 0);
+		dma_stm32_clear_ht(dma, id);
+	} else if (dma_stm32_is_tc_active(dma, id)) {
 #ifdef CONFIG_DMAMUX_STM32
 		stream->busy = false;
 #endif
 		stream->dma_callback(dev, stream->user_data, callback_arg, 0);
-	} else if (dma_stm32_is_ht_active(dma, id)) {
-		dma_stm32_clear_ht(dma, id);
+		dma_stm32_clear_tc(dma, id);
```

Comment:
The other way round didn't work with my purpose as DMA HAL needs flags to be still available when the callback is called to work as expected.
This being said, I might need to test more thoroughly other cases to be sure I didn't break anything and use this behavior only in case of "DMA channel override"

Reply:
is this still an issue?

Reply:
> is this still an issue?

For sure I can't make it work the other way round due to use of HAL API. So this is still needed, yes.

Reply:
I meant, did you still need to test more thoroughly, or are you satisfied that we can ignore the change?

Reply:
It was tested as much as I could and no issue was find. Anyway, I just protected this behavior so we should be safe on that front now.